Duplicati/Library/Main/Operation/DeleteHandler.cs

```diff
@@ -156,222 +158,233 @@ public void DoRun(Database.LocalDeleteDatabase db, ref System.Data.IDbTransactio
                         Logging.Log.WriteDryrunMessage(LOGTAG, "WouldDeleteHelp", "Remove --dry-run to actually delete files");
                 }
                 
-                if (!m_options.NoAutoCompact && (forceCompact || (toDelete != null && toDelete.Length > 0)))
+                if (!m_options.NoAutoCompact && (forceCompact || toDelete.Length > 0))
                 {
                     m_result.CompactResults = new CompactResults(m_result);
                     new CompactHandler(m_backendurl, m_options, (CompactResults)m_result.CompactResults).DoCompact(db, true, ref transaction, sharedManager);
                 }
-                
-                m_result.SetResults(
-                    from n in filesetNumbers
-                    where toDelete.Contains(n.Item2)
-                    select n, 
-                    m_options.Dryrun);
+
+                m_result.SetResults(versionsToDelete.Select(v => new Tuple<long, DateTime>(v.Version, v.Time)), m_options.Dryrun);

```

Comment:
Is versionsToDelete correct, or toDelete?

Reply:
I think `versionsToDelete` is correct.  The old implementation used the following (from line 97 of `DoRun`)
```
var filesetNumbers = db.FilesetTimes.Zip(Enumerable.Range(0, db.FilesetTimes.Count()), (a, b) => new Tuple<long, DateTime>(b, a.Value)).ToList();
```
which has local timestamps, similar to `versionsToDelete`.